packages/apex-ls/src/server/LCSAdapter.ts

```diff
@@ -213,6 +216,26 @@ export class LCSAdapter {
     );
   }
 
+  /**
+   * Utility handlers (shutdown, exit)
+   * These are registered early for proper lifecycle management
+   */
+  private setupUtilityHandlers(): void {
+    // Register shutdown handler
+    this.connection.onRequest('shutdown', (): null => {
```

Comment:
Is this part of some other pre-requisite? exit calling something makes enough sense but a dedicated shutdown request that just does nothing but log before returning null doesn't seem like it would make sense to have other than for meeting some requirement.